src/lib.rs

```diff
@@ -25,7 +25,7 @@ use std::iter::IntoIterator;
 ///
 /// Implementations of this trait can be found in
 /// [models](./model/index.html)
-pub trait Model: Clone {
```

Comment:
I removed the clone trait, since we don't clone Models anywhere in this library, and makes it hard to use Closures in Models